var/spack/repos/builtin/packages/trilinos-for-albany/package.py

```diff
@@ -0,0 +1,841 @@
+    conflicts('+teko', when='~amesos')
+    conflicts('+teko', when='~anasazi')
+    conflicts('+teko', when='~aztec')
+    conflicts('+teko', when='~ifpack')
+    conflicts('+teko', when='~ml')
+    conflicts('+teko', when='~teuchos')
+    conflicts('+teko', when='~tpetra')
+    conflicts('+tempus', when='~nox')
+    conflicts('+tempus', when='~teuchos')
+    conflicts('+tpetra', when='~kokkos')
+    conflicts('+tpetra', when='~teuchos')
+    conflicts('+zoltan2', when='~teuchos')
+    conflicts('+zoltan2', when='~tpetra')
+    conflicts('+zoltan2', when='~xpetra')
+    conflicts('+zoltan2', when='~zoltan')
+
+    conflicts('+dtk', when='~intrepid2')
+    conflicts('+dtk', when='~kokkos')
+    conflicts('+dtk', when='~teuchos')
+    conflicts('+dtk', when='~tpetra')
+    # Only allow DTK with Trilinos 12.14 and develop
+    conflicts('+dtk', when='@0:12.12.99,master')
+    conflicts('+fortrilinos', when='~fortran')
+    conflicts('+fortrilinos', when='@:99')
+    conflicts('+fortrilinos', when='@master')
+    # Can only use one type of SuperLU
+    conflicts('+superlu-dist', when='+superlu')
+    # For Trilinos v11 we need to force SuperLUDist=OFF, since only the
+    # deprecated SuperLUDist v3.3 together with an Amesos patch is working.
+    conflicts('+superlu-dist', when='@11.4.1:11.14.3')
+    # see https://github.com/trilinos/Trilinos/issues/3566
+    conflicts('+superlu-dist', when='+float+amesos2+explicit_template_instantiation^superlu-dist@5.3.0:')
+    # Amesos, conflicting types of double and complex SLU_D
+    # see
+    # https://trilinos.org/pipermail/trilinos-users/2015-March/004731.html
+    # and
+    # https://trilinos.org/pipermail/trilinos-users/2015-March/004802.html
+    conflicts('+superlu-dist', when='+complex+amesos2')
+    # PnetCDF was only added after v12.10.1
+    conflicts('+pnetcdf', when='@0:12.10.1')
+    # https://github.com/trilinos/Trilinos/issues/2994
+    conflicts(
+        '+shared', when='+stk platform=darwin',
+        msg='Cannot build Trilinos with STK as a shared library on Darwin.'
+    )
+    # ADIOS2 was only added after v12.14.1
+    conflicts('+adios2', when='@:12.14.1')
+    conflicts('+adios2', when='@xsdk-0.2.0')
+    # ###################### Dependencies ##########################
+
+    # Everything should be compiled position independent (-fpic)
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('boost', when='+boost')
+    depends_on('boost', when='+dtk')
+    depends_on('parmetis', when='+scorec')
+    depends_on('matio', when='+matio')
+    depends_on('glm')
+    depends_on('metis@5:', when='+metis')
+    depends_on('suite-sparse', when='+suite-sparse')
+    depends_on('zlib', when="+zlib")
+
+    # MPI related dependencies
+    depends_on('mpi')
+    depends_on('netcdf-c+mpi', when="~pnetcdf")
+    depends_on('netcdf-c+mpi+parallel-netcdf', when="+pnetcdf@master,12.12.1:")
+    depends_on('parallel-netcdf', when="+pnetcdf@master,12.12.1:")
```

Comment:
I think this would be clearer as:
```suggestion
    depends_on('parallel-netcdf', when="@12.12.1:+pnetcdf")
```
This is consistent with specs below and `@master` is always greater than a numerical version so it doesn't have to be specified explicitly.  With the other syntax, I thought the version was for `pnetcdf` itself, not for `trilinos`.

Reply:
I can fix this - thanks.

Reply:
Resolved.